### PR TITLE
initial react-table and allVocabs

### PR DESCRIPTION
**Pull request by zikowang**
| Status  | Type  |
| :---: | :---: |
|READY | Feature |

## Description

- [x] add table component
- [x] add table pagination
- [x] add confirm deletion dialog
- [x] add packages table screen
- [x] add groups table screen
- [x] add vocabs table screen
- [x] package delete
- [x] group delete
- [x] vocab delete
- [x] package add
- [x] group add
- [x] vocab add
- [x] package edit
- [x] group edit
- [x] vocab edit
- [x] make it look nice

## Screenshots / GIFs (if appropriate):

image
image
image
image

image
image
image
image

image
image
image
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (non-breaking change which cleans up / improves existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the **CONTRIBUTING** document.
- [x] I have considered the accessibility of my changes (i.e. did I add proper content descriptions to images, or run my changes with talkback enabled?)
- [x] I have documented my code if needed

### CODE REVIEW

File: src/Components/ConfirmDialog/ConfirmDialog.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  return (
+    <Modal size="small" title={title} open={show} onClose={onClose} xxl>
+      <div className="submit-dialog">
+        <div className="description">
+          <span>{description}</span>
+        </div>
+        <div className="actions">
+          <Button onClick={onClose} appearance="primary">
+            {cancelText ? cancelText : t("global.abort")}
```
Comment (luwol03) [inline src/Components/ConfirmDialog/ConfirmDialog.jsx:28]: couldn't we use?
```suggestion
            {cancelText || t("global.abort")}
```

File: src/Components/ConfirmDialog/ConfirmDialog.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import { useTranslation } from "react-i18next";
+
+import "./ConfirmDialog.scss";
+
+import Button from "../Button/Button";
```
Comment (luwol03) [inline src/Components/ConfirmDialog/ConfirmDialog.jsx:6]: add `.jsx` extension and also in other files for import sort plugin

File: src/Forms/PackageForm/PackageForm.jsx [hunk 3]
```diff
@@ -4,46 +4,107 @@ import { useTranslation } from "react-i18next";
...
-  const [translatedLanguage, setTranslatedLanguage] = useState("");
-  const [vocabsPerDay, setVocabsPerDay] = useState(100);
-  const [rightTranslations, setRightTranslations] = useState(2);
+  const [name, setName] = useState(defaultData ? defaultData.name : "");
+  const [foreignLanguage, setForeignLanguage] = useState(
+    defaultData
+      ? {
+          value: defaultData.foreignWordLanguage,
+          label:
+            languages.find((ele) => ele.name === "Deutsch").icon +
```
Comment (luwol03) [inline src/Forms/PackageForm/PackageForm.jsx:24]: why did you hardcode "Deutsch"?

File: src/colors.scss [hunk 4]
```diff
@@ -4,8 +4,8 @@ $color-background: #fff;
 $color-alternative: #1c1c1a;
 $color-form: #252421;
 $color-box-shadow: #212121;
-$color-main-text: #212121;
 $color-main-text-light: #383838;
+$color-main-text: #212121;
```
Comment (luwol03) [inline src/colors.scss:8]: is this change intended?

File: src/hooks/useSnack.jsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+const useSnack = () => {
+  const { showSnack } = useContext(SnackbarContext);
+
+  return {
+    showSnack,
+  };
+};
+
+export default useSnack;
```
Comment (luwol03) [inline src/hooks/useSnack.jsx:13]: can we rename this to a js file, so it doesn't sort to the components?

**Review by luwol03 — CHANGES_REQUESTED** [review]
- There is also a bug that if you edit a package and deactivate it, the state is saved, so that if you want to create a new one, it is taken over directly.

- I don't like the transition on the active state if I set e.g. group active to false in group edit dialog and then I open another dialog which has active set to true.

File: src/hooks/useSnack.jsx (hunk 5 above)
Comment (zikowang) [reply src/hooks/useSnack.jsx:13]: yes thats right

File: src/colors.scss (hunk 4 above)
Comment (zikowang) [reply src/colors.scss:8]: yes, it was not in use anyway.

File: src/Forms/PackageForm/PackageForm.jsx (hunk 3 above)
Comment (zikowang) [reply src/Forms/PackageForm/PackageForm.jsx:24]: oh, my bad

File: src/Components/ConfirmDialog/ConfirmDialog.jsx (hunk 2 above)
Comment (zikowang) [reply src/Components/ConfirmDialog/ConfirmDialog.jsx:6]: check

File: src/Components/ConfirmDialog/ConfirmDialog.jsx (hunk 1 above)
Comment (zikowang) [reply src/Components/ConfirmDialog/ConfirmDialog.jsx:28]: ok

**Comment by zikowang** [comment]
> * I don't like the transition on the active state if I set e.g. group active to false in group edit dialog and then I open another dialog which has active set to true.
We could add a loading state as a feature future PRs

**Comment by zikowang** [comment]
> * There is also a bug that if you edit a package and deactivate it, the state is saved, so that if you want to create a new one, it is taken over directly.

Fixed

**Review by luwol03 — APPROVED** [review]
